backend-rust/src/graphql_api/baker.rs

```diff
@@ -350,6 +352,124 @@ impl Baker {
+            WHERE id = true
+            "
+        )
+        .fetch_one(pool)
+        .await?;
+
+        // The `leverage_bound` and `capital_bound` are Concordium chain parameters that
+        // were set adhering to the below constraints to ensure the consensus
+        // algorithm works. We check these constraints here to ensure the values
+        // have been saved in this format to the database.
+        if current_chain_parameters.leverage_bound_numerator < 0 {
+            return Err(ApiError::InternalError(
+                "`leverage_bound_numerator` is negative in the database".to_string(),
+            ));
+        }
+        if current_chain_parameters.leverage_bound_denominator <= 0 {
+            return Err(ApiError::InternalError(
+                "`leverage_bound_denominator` is not greater than 0 in the database".to_string(),
+            ));
+        }
+        // Equality checks on floating point numbers should be avoided in general,
+        // although this seems to be within the safe zone since the leverage bound
+        // as set in the chain parameter is not close to 1 at all.
+        if (current_chain_parameters.leverage_bound_numerator as f64)
+            / (current_chain_parameters.leverage_bound_denominator as f64)
+            < 1f64
+        {
+            return Err(ApiError::InternalError(
+                "`leverage_bound` is smaller than 1 in the database".to_string(),
+            ));
+        }
+        if current_chain_parameters.capital_bound <= 0 {
+            return Err(ApiError::InternalError(
+                "`capital_bound` is not greater than 0 in the database".to_string(),
+            ));
+        }
+
+        // Calculating the `leverage_bound_cap`
+
+        // To reduce loss of precision, the value is computed in u128.
+        let leverage_bound_cap_for_pool_numerator: u128 =
+            (current_chain_parameters.leverage_bound_numerator
+                - current_chain_parameters.leverage_bound_denominator) as u128
+                * self.staked as u128;
+        // Denominator is not zero since we checked that before.
+        let leverage_bound_cap_for_pool_denominator: u128 =
+            current_chain_parameters.leverage_bound_denominator as u128;
+
+        let leverage_bound_cap_for_pool: u64 = (leverage_bound_cap_for_pool_numerator
+            / leverage_bound_cap_for_pool_denominator)
+            .try_into()
+            .unwrap_or(u64::MAX);
+        let leverage_bound_cap_for_pool: Amount = leverage_bound_cap_for_pool.into();
+
+        // Calculating the `capital_bound_cap`
+
+        let capital_bound: u128 = current_chain_parameters.capital_bound as u128;
+
+        let capital_bound_cap_for_pool: Amount = if capital_bound == 100_000u128 {
+            // To avoid dividing by 0 in the `capital bound cap` formula,
+            // we only apply the `leverage_bound_cap_for_pool` in that case.
+            leverage_bound_cap_for_pool
```

Comment:
I would be inclined to use `u64::MAX.into()` instead, because the capital cap should be independent of the leverage cap.